### DISCUSSION TITLE
toSVG returns different version than what's on the canvas

### DISCUSSION THREAD

**Discussion by kirovdone** [fabricjs/fabric.js#7626] [Q&A]
Hey,
I am facing a problem when exporting the canvas to .SVG. I figured out that i need to store all objects in .JSON in order to get a correct placement/alignment. So, I am loading .JSON file and I get the right preview on the canvas and then I save them in .JSON (returns correct version) and .SVG (need it for previews) but the function .toSVG returns a version where the design is slightly moved/displaced than the original version that I made on the canvas.

This is how I export the .SVG:
const dataUrl = this.canvas
      .toSVG({
        suppressPreamble: true,
        width: 1000,
        height: 1000,
        viewBox: {
          x: 0,
          y: 0,
          width: 1000,
          height: 1000,
        },
});

I attach a preview of one test example:
![logos-pack-104](https://user-images.githubusercontent.com/36360335/150520932-bcf81a39-5c4f-401c-b157-e604ea57fc01.png)

Has someone an idea how I can fix the issue and is it an internal bug on fabric.js/canvas or there are some options that I can include to make it right?

Thanks.

**Comment by asturur** [comment]
Oh that is weird, can you actually replicate it with code in a fiddle so that we can see it happening?

**Reply by kirovdone** [reply]
I am not sure I can do it as the canvas and all functionalities are implemented in React.js project. I am creating a custom .SVG and importing on the canvas and then i export in .JSON and import again. The canvas preview is perfectly accurate but when I export in svg using canvas.toSVG, the objects are moved. Really weird, if someone want to take a look, I am open to share my screen to explain and show what exactly the issue is.

**Reply by asturur** [reply]
it is a matter of dedicating time to it.
you need to have a fiddle with something on the canvas, that when exported to SVG looks different.
All the rest is meaningless.
If you can't make the error happed in a fiddle/codepen that is a strong hint that the error is in your custom code.

**Reply by kirovdone** [reply]
Yeah, the issue could be in my code. That's why i placed it in discussion so if anyone with more experience is open to check and help me as this is my first startup project working with canvas and it is really important to fix it asap.

**Reply by asturur** [reply]
did you try not specifying the viewbox? what happens?

**Reply by kirovdone** [reply]
I did:

const dataUrl = this.canvas
.toSVG({
suppressPreamble: true,
width: 1000,
height: 1000,
viewBox: {
x: 0,
y: 0,
width: 1000,
height: 1000,
},
});

**Reply by asturur** [reply]
if you specify viewbox manually the object will be moved around depedning on the coordinates they have.
The viewbox is exactly like the viewportTransform in fabric, and it is going to define where you see the objects in the svg.
Why do you need to set it manually?
